pyscf/grad/casscf.py

```diff
@@ -51,24 +56,34 @@ def grad_elec(mc_grad, mo_coeff=None, ci=None, atmlst=None, verbose=None):
     nao, nmo = mo_coeff.shape
     nao_pair = nao * (nao+1) // 2
 
+    # Necessary kludge because gfock isn't zero in occ-virt space in SA-CASSCf
+    # Among many other potential applications!
+    if hasattr (mc, '_tag_gfock_ov_nonzero'):
+        if mc._tag_gfock_ov_nonzero:
+            nocc = nmo
```

Comment:
Why nocc is condtionally set to nmo? Based on the implementations below, I think it's safe to remove nocc and use nmo to replace it, right?

Reply:
Well, yes, except this causes some of the tests of the original single-state CASSCF gradient to fail (with error << value but > the threshold of the test scripts) because of the finite convergence threshold of CASSCF iteration. I made it conditional for the sake of strict backwards-compatibility, in other words.